### ISSUE TITLE
Mongock use the wrong class loader when used with Quarkus

### ISSUE THREAD

**Issue by tms0** [mongock/mongock#663]
## Description

When setting Mongock with`addMigrationClasses` in a Quarkus project, Mongock does not find the given class.

## PRIORITY

CRITICAL

### Mongock
- Mongock version: 5.4.0
- Modules involved: standalone

### Environment
- Framework and libraries versions: Quarkus v3.8.2

## Steps to Reproduce

See this reproductive example: https://github.com/tms0/quarkus-mongock-issue

## Behaviour

**Expected behavior:** The given change set should be executed

**Actual behavior:** No Change set detected

**How often the bug happens:** 100%

## Additional context

Quarkus use a custom [Runtime ClassLoader](https://quarkus.io/guides/class-loading-reference), while Mongock use the system ClassLoader to read the migration classes (https://github.com/mongock/mongock/blob/master/mongock-core/mongock-runner/mongock-runner-core/src/main/java/io/mongock/runner/core/executor/ChangeExecutorBase.java#L474).

I will submit a PR soon to use the ClassLoader of the current thread instead of the system one, it should work everywhere.

**Comment by dieppa** [comment]
Thanks @tms0 . I will review the PR once it's raised :)

**Comment by janknobloch** [comment]
@tms0 worked fine for me in regards to the classlolder - no touches needed there - yet i had to manually force 

```<dependency>
<groupId>org.reflections</groupId>
<artifactId>reflections</artifactId>
<version>0.10.1</version>
</dependency>

as the quarkus bom forces a downgrade onto 0.9.x which broke my integration, yet im not running native - are you?

**Comment by dieppa** [comment]
Thanks @janknobloch, `reflections` library in some of its versions, produces this error. Not sure this is the fix for native image. If @tms0 can you confirm this is(or not) a fix, would be great. Otherwise, as said, I will wait for the PR

**Comment by janknobloch** [comment]
@dieppa - sadly haven't had the time to test native for now - hope to find some time on the weekend to report back on this if my extension works at least fine in native (with the given fix) + I need to build most likely some more build steps with "keep migration files / jandex etc"

@tms0 let me know if I can be of any help,

so far my extension runs fine with the upper mentioned quick fix, if anyone has a nicer suggestion on the reflections issue e.g. maybe to overwrite the quarkus bom somehow implicitly from the extension itself and if that's the proper way on dealing with this I'm happy for any input :-)

**Comment by tms0** [comment]
Both JVM and native modes are impacted by this issue.

@janknobloch I guess you use the package scan method ? It should explain why you don't have this issue. But package scan will not work in native mode, because `org.reflections` use `org.jboss.vfs`, which is not supported by GraalVM (if I understand correctly).

**Comment by dieppa** [comment]
So, @tms0 , Correct me if I am wrong, but I understand you are planning to raise a PR addressing this, right?

**Comment by janknobloch** [comment]
@tms0 yes indeed im using the packageScan feature for now, I think when using addMigrationClasses you would need to search during build time for the given mongock annotations and use a buildstep in conjunction with unremovableOf to keep those files in the class path?

**Comment by tms0** [comment]
> So, @tms0 , Correct me if I am wrong, but I understand you are planning to raise a PR addressing this, right?

Here you are : https://github.com/mongock/mongock/pull/664

**Comment by dieppa** [comment]
@tms0 Approved pipeline run 😉

**Comment by dieppa** [comment]
Sorry I didn't let you know, but the PR was merged and released in version `5.4.1`
